src/modal.tsx

```diff
@@ -241,6 +250,16 @@ export class ReactNativeModal extends React.Component<ModalProps, State> {
     this.animationIn = animationIn;
     this.animationOut = animationOut;
 
+    this.scrollOffset = this.props.scrollOffset || defaultProps.scrollOffset;
+    this.panResponderThreshold =
+      this.props.panResponderThreshold || defaultProps.panResponderThreshold;
+    this.scrollOffsetMax =
+      this.props.scrollOffsetMax ?? defaultProps.scrollOffsetMax;
+    this.swipeThreshold =
+      this.props.swipeThreshold ?? defaultProps.swipeThreshold;
+    this.onModalShow = this.props.onModalShow ?? defaultProps.onModalShow;
+    this.onModalHide = this.props.onModalHide ?? defaultProps.onModalHide;
```

Comment:
I might be missing something here, but by using instance variables instead of props it means that these values won't be updated if the props are changed, right?

Reply:
You're right. That was my bad.I just pushed a commit fixing this and adding nullish coalescing operator to fix TS errors